[IMP] product_barcodelookup: product attribute improvements

sjai-odoo opened the pull request:
- We have given visibility to Age Group and Manufecturer attributes
created in product_barcodelookup.

Related: https://github.com/odoo/enterprise/pull/63740/
task- 3965079

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/167807).

Feyensv commented:
> * We have created some of the necessary attributes, enabling their automatic creation when they are received from barcode lookup.

What do you mean by necessary ?
The new records you are creating are only in demo data files, so they probably aren't necessary :thinking:.
Also, if they are used by the barcodelookup module, why don't you specify it in the module directly?

sjai-odoo commented:
> > * We have created some of the necessary attributes, enabling their automatic creation when they are received from barcode lookup.
> 
> What do you mean by necessary ? The new records you are creating are only in demo data files, so they probably aren't necessary 🤔. Also, if they are used by the barcodelookup module, why don't you specify it in the module directly?

Hi @Feyensv ,

I'm sorry for the mistake. I first put the data in the demo files by accident, but now I've moved it to the data file of the product_barcodelookup module. I've also updated the commit message to make it clearer.

Thanks.

Feyensv commented on addons/website_sale/data/data.xml:
```diff
@@ -22,6 +22,12 @@
             <field name="name">Brand</field>
             <field name="sequence">0</field>
         </record>
+        <record id="product_barcodelookup.product_attribute_5" model="product.attribute">
+            <field name="visibility">hidden</field>
+        </record>
+        <record id="product_barcodelookup.product_attribute_6" model="product.attribute">
+            <field name="visibility">hidden</field>
+        </record>
```
Sadly, you cannot reference a record which is not defined in the module dependencies (does it really matter to hide those though) ?

sjai-odoo replied:
I have changed the code accordingly now we will create these attributes when encountered but not available.

sjai-odoo commented:
Handled this in [enterprise](https://github.com/odoo/enterprise/pull/63740), so closing this PR.
